File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking2010.Views
Imports DevExpress.XtraBars.Docking2010.Views.WindowsUI

Namespace WindowsFormsApplication1
	Public Partial Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			pageGroup1.Properties.ShowPageHeaders = DevExpress.Utils.DefaultBoolean.False
			AddHandler windowsUIView1.QueryDocumentActions, AddressOf windowsUIView1_QueryDocumentActions
			AddHandler windowsUIView1.TileClick, AddressOf windowsUIView1_TileClick
		End Sub
		Private Sub windowsUIView1_TileClick(ByVal sender As Object, ByVal e As TileClickEventArgs)
            If e.Tile Is document1Tile Then
                page1.Document = document1
            End If
            If e.Tile Is document2Tile Then
                page1.Document = document2
            End If
		End Sub
		Private Sub windowsUIView1_QueryControl(ByVal sender As Object, ByVal e As QueryControlEventArgs) Handles windowsUIView1.QueryControl
			If e.Document Is document1 Then
				e.Control = New Views.ucModuleA(New ViewModels.ViewModelModuleA())
			End If
			If e.Document Is document2 Then
				e.Control = New Views.ucModuleB(New ViewModels.ViewModelModuleB())
			End If
		End Sub
		'
        Private Sub windowsUIView1_QueryDocumentActions(sender As Object, e As QueryDocumentActionsEventArgs)
            e.DocumentActions.Add(New DocumentAction(Sub(doc) Hello(e.Document)) With { _
                 .Caption = "Hello" _
            })
            If e.Source Is pageGroup1 Then
                e.DocumentActions.Add(New DocumentAction(Function(doc) doc Is document2, Sub(doc) windowsUIView1.ActivateDocument(document1)) With { _
                     .Caption = "Document 1" _
                })
                e.DocumentActions.Add(New DocumentAction(Function(doc) doc Is document1, Sub(doc) windowsUIView1.ActivateDocument(document2)) With { _
                     .Caption = "Document 2" _
                })
            End If
        End Sub
		Private Sub Hello(ByVal document As Document)
			MessageBox.Show("Hello from " & document.Caption)
		End Sub
	End Class
End Namespace

File: CS/WindowsFormsApplication1/Form1.cs
using System.Windows.Forms;
using DevExpress.XtraBars.Docking2010.Views;
using DevExpress.XtraBars.Docking2010.Views.WindowsUI;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            pageGroup1.Properties.ShowPageHeaders = DevExpress.Utils.DefaultBoolean.False;
            windowsUIView1.QueryDocumentActions += windowsUIView1_QueryDocumentActions;
            windowsUIView1.TileClick += windowsUIView1_TileClick;
            windowsUIView1.AllowCaptionDragMove = DevExpress.Utils.DefaultBoolean.True;
        }
        void windowsUIView1_TileClick(object sender, TileClickEventArgs e) {
            if(e.Tile == document1Tile)
                page1.Document = document1;
            if(e.Tile == document2Tile)
                page1.Document = document2;
        }
        void windowsUIView1_QueryControl(object sender, QueryControlEventArgs e) {
            if(e.Document == document1)
                e.Control = new Views.ucModuleA(new ViewModels.ViewModelModuleA());
            if(e.Document == document2)
                e.Control = new Views.ucModuleB(new ViewModels.ViewModelModuleB());
        }
        //
        void windowsUIView1_QueryDocumentActions(object sender, QueryDocumentActionsEventArgs e) {
            e.DocumentActions.Add(new DocumentAction(Hello) { Caption = "Hello" });
            if (e.Source == pageGroup1) {
                e.DocumentActions.Add(new DocumentAction((doc) => doc == document2,
                    (doc) => windowsUIView1.ActivateDocument(document1)) { Caption = "Document 1" });
                e.DocumentActions.Add(new DocumentAction((doc) => doc == document1,
                    (doc) => windowsUIView1.ActivateDocument(document2)) { Caption = "Document 2" });
            }
        }
        void Hello(Document document) {
            MessageBox.Show("Hello from " + document.Caption);
        }
    }
}